grafana-plugin/src/containers/RemoteSelect/RemoteSelect.tsx

```diff
@@ -62,17 +68,23 @@ const RemoteSelect = inject('store')(
 
     const [options, setOptions] = useReducer(mergeOptions, []);
 
-    useEffect(() => {
-      makeRequest(href, {}).then((data) => {
-        setOptions(getOptions(data.results || data));
-      });
+    const loadOptionsCallback = useCallback(async (query?: string): Promise<SelectableValue[]> => {
+      try {
+        const data = await makeRequest(href, { params: { search: query } });
+        const options = getOptions(data.results || data);
+        setOptions(options);
+
+        return options;
+      } catch (e) {
+        if (axios.isAxiosError(e) && e.response.status === 403 && requiredUserAction) {
```

Comment:
can we omit checking `axios.isAxiosError(e)` here? because RemoteSelect shouldn't know which library we use for network requests

Reply:
^ I thought the same about this one, but Joey mentioned he encountered some type error there :thinking:

Reply:
I just double checked this morning:
```tsx
try {
  // makeRequest()
} catch (e) {
}
```

@teodosii you were right, `e` is typed as `any` here so technically accessing `e.response.status` is _okay_.

I like your point about the component not having to know about `axios`, but I also think a type-guard can be useful. I'd propose adding a custom type-guard to `src/network/index.ts` that wraps `axios.isAxiosError` (I'd also update the few other spots in the codebase where we use this function).